src/column.rs

```diff
@@ -361,26 +402,65 @@ impl Column {
 	}
 }
 
+pub fn packed_node_size(data: &Vec<u8>, num_children: u8) -> usize {
+	1 + data.len() + num_children as usize * 8
+}
+
+pub fn pack_node_data(data: Vec<u8>, child_data: Vec<u8>, num_children: u8) -> Vec<u8> {
```

Comment:
I’ve improved this a bit. To do a truncate of the vector I think the various get functions that the node data comes from will need to return a mut Vec value. Is that correct? It would be possible to do, though I haven’t yet.